v2/smux.go

```diff
@@ -0,0 +1,44 @@
+package cmux
+
+import (
+	"net"
+
+	"github.com/xtaci/smux"
+)
+
+func translateSmuxErr(err error) error {
+	if err == smux.ErrTimeout {
+		return ErrTimeout
+	} else {
+		return err
+	}
+}
+
+type smuxSession struct{ session *smux.Session }
+
+func (s *smuxSession) AcceptStream() (net.Conn, error) { return s.session.AcceptStream() }
+func (s *smuxSession) OpenStream() (net.Conn, error)   { return s.session.OpenStream() }
+func (s *smuxSession) Close() error                    { return s.session.Close() }
+func (s *smuxSession) NumStreams() int                 { return s.session.NumStreams() }
+
+type smuxProtocol struct {
+	smux.Config
+}
+
+func NewSmuxProtocol() *smuxProtocol {
+	return &smuxProtocol{*smux.DefaultConfig()}
```

Comment:
Should we allow overriding the config here? Or how could one customize smux?

Reply:
As-is you customize via the exported struct fields.
```
mySmux := NewSmuxProtocol()
mySmux.Version = 2
...
```

But could make this more one-liner-ish by allowing it to accept an smux.Config here.

Reply:
I made the NewSmuxProtocol just take an `smux.Config` and hid it inside, better to assume you can't continue to change it after it's created I think.